src/client/Client.cpp

```diff
@@ -3621,3 +3621,17 @@ bool Client::verifyVolumeID(const std::string& id){
 		return false;
 	return true;
 }
+
+bool Client::verifyNamespaceName(const std::string& namespaceName){
+	if (namespaceName.length() <= LABEL_NAME_MAX_LEN){
+		if (std::iswalnum(namespaceName.front()) && std::iswalnum(namespaceName.back())) {
```

Comment:
These checks will involve invalid operations if `namespaceName` is empty. It would make sense to add a check right at the start of the function to reject empty names. (You could also roll the maximum length check into this, and by making the check immediately return `false` for bad cases you can save a level of indentation for the meat of this function.) It looks like as the function is being called now it won't get empty inputs, but this is a very cheap check to prevent trouble down the road if someone starts calling it from a new context, etc.

Reply:
You can apply the same transfomation to the condition with the `iswalnum` checks to un-nest the contents from it as well: negate the condition and immediately return false if that is met.

Reply:
Does [this](https://github.com/slateci/slate-client-server/pull/171/commits/9bc330263f81e3e022c8cedaa835027099ae0449#diff-e06cb3ee018bdf21480a0079e3248c0c0a11d89c57ad4db86f7d8a847e23a580R3627-R3629) seem okay?

Reply:
That works, although note that none of the `else`s following an `if` which contains a `return` are needed.

Reply:
Makes sense, got rid of unnecessary `else`s in [this](https://github.com/slateci/slate-client-server/pull/171/commits/e9a695b2b74392374c135b699eabd81b97c70f4b) commit